armi/reactor/converters/axialExpansionChanger.py

```diff
@@ -488,96 +508,130 @@ def _getLinkedComponents(self, b, c):
         RuntimeError
             multiple candidate components are found to be axially linked to a component
         """
-        lstLinkedC = [None, None]
+        self.linkedComponents[c] = collections.defaultdict(list)
         for ib, linkdBlk in enumerate(self.linkedBlocks[b]):
             if linkdBlk is not None:
                 for otherC in getSolidComponents(linkdBlk.getChildren()):
-                    if _determineLinked(c, otherC):
-                        if lstLinkedC[ib] is not None:
-                            errMsg = (
-                                "Multiple component axial linkages have been found for "
-                                f"Component {c}; Block {b}; Assembly {b.parent}."
-                                " This is indicative of an error in the blueprints! Linked components found are"
-                                f"{lstLinkedC[ib]} and {otherC}"
-                            )
-                            runLog.error(msg=errMsg)
-                            raise RuntimeError(errMsg)
-                        lstLinkedC[ib] = otherC
-
-        self.linkedComponents[c] = lstLinkedC
-
-        if lstLinkedC[0] is None:
+                    if AssemblyAxialLinkage._determineLinked(c, otherC):
+                        self.linkedComponents[c][ib].append(otherC)
+
+        if not self.linkedComponents[c][0]:
             runLog.debug(
                 f"Assembly {self.a}, Block {b}, Component {c} has nothing linked below it!",
                 single=True,
             )
-        if lstLinkedC[1] is None:
+        if not self.linkedComponents[c][1]:
             runLog.debug(
                 f"Assembly {self.a}, Block {b}, Component {c} has nothing linked above it!",
                 single=True,
             )
 
+    @staticmethod
+    def _determineLinked(componentA, componentB) -> bool:
+        """Determine axial component linkage for two solid components.
 
-def _determineLinked(componentA, componentB):
-    """Determine axial component linkage for two components.
+        Parameters
+        ----------
+        componentA : :py:class:`Component <armi.reactor.components.component.Component>`
+            component of interest
+        componentB : :py:class:`Component <armi.reactor.components.component.Component>`
+            component to compare and see if is linked to componentA
 
-    Parameters
-    ----------
-    componentA : :py:class:`Component <armi.reactor.components.component.Component>`
-        component of interest
-    componentB : :py:class:`Component <armi.reactor.components.component.Component>`
-        component to compare and see if is linked to componentA
+        Notes
+        -----
+        If componentA and componentB are both solids and the same type, geometric overlap can be checked via
```

Comment:
I think your change lost the check on whether the components are solids or not. Am I wrong about that?